unit_tests/src/test_resource_administration.cpp

```diff
@@ -77,6 +80,33 @@ TEST_CASE("resource administration")
         REQUIRE(info->host_name() == ufs_info.host_name);
     }
 
+    // see issue #5170
+    SECTION("verify resource id from json serialization")
+    {
+        // we need a fresh connection becuase the cache is probably out of date
+        auto conn_pool = irods::make_connection_pool();
+        auto conn = conn_pool->get_connection();
+
+        auto [ec, info] = adm::client::resource_info(conn, ufs_info.resource_name);
+        REQUIRE(ec.value() == 0);
+        REQUIRE(info->type() == ufs_info.resource_type);
+
+        bytesBuf_t *jbuf;
+        REQUIRE(rcServerReport((rcComm_t*)conn, &jbuf) == 0);
+        std::string report_str((char*)jbuf->buf, jbuf->len);
+
+        auto report = nlohmann::json::parse(report_str);
+        REQUIRE(report["resources"].size() >= 2);
+        std::string id_str;
+        for (auto &resource : report["resources"]) {
```

Comment:
`resource` can be marked `const` as well.  this can help the compiler produce more optimized code.